packages/ui/src/util/hooks/useStateLogs.ts

```diff
@@ -0,0 +1,47 @@
+import { ResponseGetState } from "@block-wallet/background/utils/types/communication"
+import { useBlankState } from "../../context/background/backgroundHooks"
+
+/**
+ * A hook that returns a function that downloads the state logs (excluding sensitive data)
+ *
+ * @returns {downloadStateLogsHandler} - A function that downloads the state logs
+ */
+const useStateLogs = () => {
+    const state = useBlankState()!
+    const downloadStateLogsHandler = () => {
+        let stateLogs: Partial<ResponseGetState> = state
+        // Remove user sensitive data from state logs
+        delete stateLogs.hiddenAccounts
+        delete stateLogs.antiPhishingImage
+        delete stateLogs.addressBook
+        delete stateLogs.recentAddresses
+
+        delete stateLogs.previousWithdrawals
+        delete stateLogs.pendingDeposits
+        delete stateLogs.depositsCount
+        delete stateLogs.pendingWithdrawals
+        delete stateLogs.areDepositsPending
+        delete stateLogs.areWithdrawalsPending
+        delete stateLogs.isVaultInitialized
+        delete stateLogs.isImportingDeposits
+        delete stateLogs.importingErrors
+
+        const stateJsonString = `data:text/json;charset=utf-8,${encodeURIComponent(
+            JSON.stringify(stateLogs)
+        )}`
+
+        const fileName = "StateLogs"
+
+        const link = document.createElement("a")
```

Comment:
Shouldn't we remove the anchor element after downloading?

Reply:
I think it's okay, as I didn't attach it to the body.
I can use link.remove() though I am not sure if it might cause issues as I can't await for the link.click() function

Reply:
Yes, its true. You are not attaching the element to the DOM. 👍